### ISSUE TITLE
feature request: support 'oneof-anyof-allof-not'

### ISSUE THREAD

**Issue by amitport** [nestjs/swagger#43]
https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/

please consider

thanks!

**Comment by spreadingmind** [comment]
As far as I understand this feature requires migrating from swagger 2.0 currently used by @nest/swagger to openapi 3

**Comment by hlohrenz** [comment]
I need this in a lot of my response case scenarios :( I'm returning a payload similar to this:

```json
{
  "items": Item[],
  "articles": NewsArticle[]
}
```

Any chance this is going to be upgraded?

**Comment by hlohrenz** [comment]
For anybody wanting a workaround for now so your codegen still detects it to generate a typescript file for it for your front-end, I used this code:

```ts
export class HomePayload {

  @ApiModelProperty({
    isArray: true
  })
  items: Object; // This will eventually change to an Entity

  @ApiModelProperty({
    isArray: true
  })
  articles: NewsArticle;

}
```

Then in my controller, I just have this:
```ts
@ApiOkResponse({
  type: HomePayload,
  description: 'Returns payload for homepage.'
})
@Get('home')
home() {
  // Code here
}
```

Then my Swagger documentation looks like this:
![image](https://user-images.githubusercontent.com/34523008/48506592-cc0b8900-e817-11e8-91d2-8c67d2860d07.png)

**Comment by Epenance** [comment]
Any updates on this? Running into the same issue in a lot of places of our swagger type generations

**Comment by vkartaviy** [comment]
It would be great to have this for `@ApiResponse` also...

**Comment by seblemeur** [comment]
Really needed

**Comment by Digister** [comment]
Is there a rough estimation, when this will be available?

**Comment by kamilmysliwiec** [comment]
Available in 4.0.0
https://docs.nestjs.com/recipes/swagger#oneof-anyof-allof

You can use the same approach with `@ApiBody()`, `@ApiQuery()` etc etc

**Comment by amitport** [comment]
Thanks! @kamilmysliwiec , BTW completely unrelated-note that the auto-scroll to hash (`#oneof-anyof-allof`) does not work for me (it actually goes to `#circular-dependencies`)
